catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -59,55 +60,72 @@
 
     private int subDirLevels;
 
+    private PeriodicBatchExecutor<Metacard> periodicDrain;
+
     private boolean enableBackupPlugin = true;
 
     public CatalogBackupPlugin() {
+
+        init();
+    }
+
+    private void init() {
         subDirLevels = 0;
+        setPeriodicDrain(new PeriodicBatchExecutor<>(1000, 60, TimeUnit.SECONDS));
+        getPeriodicDrain().execute(this::backup);
     }
 
     /**
      * Backs up created metacards to the file system backup.
      *
      * @param input the {@link CreateResponse} to process
-     * @return
+     * @return {@link CreateResponse}
      * @throws PluginExecutionException
      */
     @Override
-    public CreateResponse process(CreateResponse input) throws PluginExecutionException {
+    public synchronized CreateResponse process(CreateResponse input)
```

Comment:
Yes, I was under the impression that that is exactly what we were doing. We don't need to fix the backup plugin only for ingest, we need to fix it for updating and deleting as well. Those operations are seeing the same performance hit that ingest is seeing because of this. There are a few different ways you could do this. I think you can assume that you're never going to get an update or delete response for a metacard that has yet to be created, because the rest of the framework is going to protect you against that. You should also be able to assume that create, update, and delete operations will actually hit this plugin in the correct order. At that point it is simply a matter of queueing them up as you work them off in your single thread.